src/layout/DwindleLayout.cpp

```diff
@@ -108,11 +108,18 @@ void CHyprDwindleLayout::applyNodeDataToWindow(SDwindleNodeData* pNode, bool for
     const bool DISPLAYTOP    = STICKS(pNode->position.y, PMONITOR->vecPosition.y + PMONITOR->vecReservedTopLeft.y);
     const bool DISPLAYBOTTOM = STICKS(pNode->position.y + pNode->size.y, PMONITOR->vecPosition.y + PMONITOR->vecSize.y - PMONITOR->vecReservedBottomRight.y);
 
-    const auto PBORDERSIZE = &g_pConfigManager->getConfigValuePtr("general:border_size")->intValue;
-    const auto PGAPSIN     = &g_pConfigManager->getConfigValuePtr("general:gaps_in")->intValue;
-    const auto PGAPSOUT    = &g_pConfigManager->getConfigValuePtr("general:gaps_out")->intValue;
-
     const auto PWINDOW = pNode->pWindow;
+    // get specific gaps and rules for this workspace,
+    // if user specified them in config
+    const auto WORKSPACE_RULE = g_pConfigManager->getWorkspaceRuleFor(g_pCompositor->getWorkspaceByID(PWINDOW->m_iWorkspaceID));
+
+    auto       gapsIn     = WORKSPACE_RULE.gapsIn.value_or(g_pConfigManager->getConfigValuePtr("general:gaps_in")->intValue);
+    auto       gapsOut    = WORKSPACE_RULE.gapsOut.value_or(g_pConfigManager->getConfigValuePtr("general:gaps_out")->intValue);
+    auto       borderSize = WORKSPACE_RULE.borderSize.value_or(g_pConfigManager->getConfigValuePtr("general:border_size")->intValue);
```

Comment:
pointers to config values not static

Reply:
How would you do that though? I'm not very well versed in C++ unfortunately, and the problem here would be that `WORKSPACE_RULE` is not static (and neither its members). But again, I'm not that expert with C++ and I might be missing something here.

Reply:
```cpp
auto       gapsIn     = WORKSPACE_RULE.gapsIn.value_or(g_pConfigManager->getConfigValuePtr("general:gaps_in")->intValue);
```

```cpp
static auto* const PGAPSIN = &g_pConfigManager->getConfigValuePtr("general:gaps_in")->intValue;
auto                         gapsIn    = WORKSPACE_RULE.gapsIn.value_or(*PGAPSIN);
```